packages/signals/src/produce-selectors.ts

```diff
@@ -0,0 +1,27 @@
+import { Signal, inject } from '@angular/core';
+import { Store, TypedSelector, ɵSelectorReturnType } from '@ngxs/store';
+
+import { RequireAtLeastOneProperty } from './types';
+
+export type SelectorMap = Record<string, TypedSelector<unknown>>;
+
+export function produceSelectors<T extends SelectorMap>(
+  selectorMap: RequireAtLeastOneProperty<T>
+) {
+  const store = inject(Store);
+
+  return Object.entries(selectorMap).reduce((accumulator, [key, selector]) => {
+    Object.defineProperty(accumulator, key, {
+      value: store.selectSignal(selector)
+    });
```

Comment:
Great, but I was proposing that we implement a getter that calls the signal.
This would remove the need to use the brackets to get the value.

Reply:
do you mean the following?

```ts
const signal = store.selectSignal(selector);

defineProp(object, key, {
  get: () => signal()
})
```

Reply:
I believe this would deviate from the typical approach of using signals, especially in templates, where signals are typically accessed using brackets to retrieve their values.

Reply:
Yes, I was meaning that, but now I'm thinking that get wrapping behaviour could be better acheived as another utility (maybe a future discussion).